flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/FileStorePathFactory.java

```diff
@@ -106,7 +114,12 @@ public Path toTmpSnapshotPath(long id) {
     }
 
     public DataFilePathFactory createDataFilePathFactory(BinaryRowData partition, int bucket) {
-        return new DataFilePathFactory(root, getPartitionString(partition), bucket);
+        return createDataFilePathFactory(partition, bucket, formatIdentifier);
+    }
+
+    public DataFilePathFactory createDataFilePathFactory(
```

Comment:
Why introduce this method? There is a `formatIdentifier` in here.

Reply:
Sorry, I submitted the wrong one. Has been updated.